src/kOS.Safe/Compilation/Opcode.cs

```diff
@@ -893,7 +892,17 @@ public override void Execute(ICpu cpu)
             else if (value is double)
                 result = -((double)value);
             else
-                throw new KOSUnaryOperandTypeException("negate", value);
+            {
+                // Generic last-ditch to catch any sort of object that has
+                // overloaded the unary negate operator '-'.
+                // (For example, kOS.Suffixed.Vector and kOS.Suffixed.Direction)
+                Type t = value.GetType();
+                MethodInfo negateMe = t.GetMethod("op_UnaryNegation", BindingFlags.Static | BindingFlags.Public); // C#'s alternate name for '-' operator
+                if (negateMe != null)
+                    result = negateMe.Invoke(null, new[]{value}); // value is an arg, not the 'this'.  (Method is static.)
```

Comment:
I did it this way because I didn't want to have a Vector and Direction Interface in kOS.Safe when the public methods of them contain mention of Unity classes like ```Vector3``` and KSP classes like ```Vector3D```.

Therefore I picked a solution in which ANY negatable object works without knowing what its type is at compile time.  In general I think it's a more elegant solution anyway, as it's more future-compatible that way.